// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

message Frame {
    uint64 pc = 1;
    string file_name = 2;
    string function_name = 3;
}

message AllocEvent {
    uint64 addr = 1;
    uint32 size = 2;
    repeated Frame frame_info = 3;
}

message FreeEvent {
    uint64 addr = 1;
    repeated Frame frame_info = 2;
}

message NativeHookData {
    uint64 tv_sec = 1;
    uint64 tv_nsec = 2;
    oneof event {
        AllocEvent alloc_event = 3;
        FreeEvent free_event = 4;
    }
}

message BatchNativeHookData {
    repeated NativeHookData events = 1;
}